ledger/src/blockstore_meta.rs

```diff
@@ -101,22 +104,38 @@ impl Index {
 
 impl ShredIndex {
     pub fn num_shreds(&self) -> usize {
-        self.index.len()
+        self.num_present
     }
 
-    pub fn present_in_bounds(&self, bounds: impl RangeBounds<u64>) -> usize {
-        self.index.range(bounds).count()
+    pub fn present_in_bounds(&self, range: Range<u64>) -> usize {
+        let mut count = 0;
+        // We should only search in the overlap of the index's values and range
+        for idx in cmp::min(range.start, self.index.len())..cmp::min(range.end, self.index.len()) {
```

Comment:
With the BitVector, we have to read each bit in the range. While this isn't ideal, I don't think it is a huge issue since practically speaking, we only call this function with [a range the same size of FEC sets](https://github.com/solana-labs/solana/blob/master/ledger/src/blockstore_meta.rs#L204-L207).